content/en/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning.md

```diff
@@ -144,6 +143,115 @@ version sort order is `v1`, followed by `v1beta1`. This causes the kubectl
 command to use `v1` as the default version unless the provided object specifies
 the version.
 
+## Webhook conversion
+
+The above example has a None converter between versions which only set `apiVersion` field
+on conversion and do not change the rest of the object. API server supports webhook conversion that
+calls an external service in case of a conversion is required, e.g. when:
+
+* CR is requested in a different version than stored version.
+* CR list request contains objects of different versions than the request version.
```

Comment:
sounds like a list can have different versions. It can't.

Reply:
I think this is in reference to https://github.com/kubernetes/kubernetes/pull/67006#discussion_r232415232

that is an internal implementation detail of storage, and I wouldn't call it out here. `A custom resource is requested in a different version than stored version.` sufficiently covers that case.

Reply:
> sounds like a list can have different versions. It can't.

Yeah, it really can. @liggitt I disagree. This detail is exposed to webhook author by sending them a list of different versions.

Reply:
We don't send them a MyKindList containing items of different versions. The conversion request we send is not a custom resource list, and the lists served from the API server have uniformly versioned items by the time they exit the API server.

Reply:
We should make clear that the conversion request can contain items in different versions that should all be converted to the requested version, but I think describing the scenario as "CR list request contains objects of different versions" is more confusing than helpful.

Reply:
I see your point. Will do.